### PR TITLE
SMS forms

### PR DESCRIPTION
**Pull request by alxndrsn**
Issue: #4812

### CODE REVIEW

File: webapp/src/js/services/form2sms.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+          $log.error('Form2Sms failed: ' + err);
+          throw err;
+        });
+    };
+  });
+
+const concat = (...args) => args.join('');
+const spaced = (...args) => args.join(' ');
+
+function bitfield() {
```
Comment (SCdF) [inline webapp/src/js/services/form2sms.js]: There is no test of this

File: webapp/src/js/services/enketo.js [hunk 2]
```diff
@@ -530,6 +532,54 @@ angular.module('inboxServices').service('Enketo',
       form.output.update();
     };
 
+    // REVIEW this is quite big and unrelated to enketo itself; maybe it should
+    // be a separate service?  Or combined into the current Form2sms service as
+    // e.g. FormsAsSms service?
+    function submitFormBySmsIfApplicable(doc) {
```
Comment (SCdF) [inline webapp/src/js/services/enketo.js]: Agree, this should be a separate service. You have:
 - Enketo, which handles bootstrapping and parsing results from that beast
 - This, which handles extra SMS sending logic
 - FormsAsSms, which handles the api understanding and parsing of how to convert a form into an SMS

File: webapp/src/js/services/form2sms.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    'use strict';
+    'ngInject';
+
+    return function(doc) {
+      if(!doc) {
+        return Promise.reject(new Error('No doc provided.'));
+      }
+
+      return DB()
+        .get(`form:${doc.form}`)
```
[the remark this answers is not in the document]
Comment (alxndrsn) [reply webapp/src/js/services/form2sms.js]: Only the transformed XML is cached: https://github.com/medic/medic-webapp/blob/master/webapp/src/js/services/enketo.js#L129-L134

File: webapp/src/js/services/form2sms.js (hunk 1 above)
Comment (alxndrsn) [reply webapp/src/js/services/form2sms.js]: Removed.

Comment (alxndrsn) [reply webapp/src/js/services/form2sms.js]: It's now been removed.

**Pull request closed without merging**